README.md

```diff
@@ -464,4 +476,39 @@ Prevents a function from running if it's state has `processing` set to true at t
 Flows together `status`, `clearStatus`, `concurrency`, and `error`, taking `extend` and `timeout` as optional parameters to construct the aspect
 
 #### deprecate
-Utility for marking functions as deprecated - it's just a `before` with a console.warn. Takes the name of thing being deprecated, optionally deprecation version, and optionally an alternative and returns a higher order function which you can wrap deprecated methods in. This is what's used internally to mark deprecations. Includes a partial stack trace as part of the deprecation warning.
\ No newline at end of file
+Utility for marking functions as deprecated - it's just a `before` with a console.warn. Takes the name of thing being deprecated, optionally deprecation version, and optionally an alternative and returns a higher order function which you can wrap deprecated methods in. This is what's used internally to mark deprecations. Includes a partial stack trace as part of the deprecation warning.
+
+## Trees
+All tree functions take a traversal function so that you can customize how to traverse arbitrary nested structures.
+
+*Note*: Be careful about cyclic structures that can result in infinite loops, such as objects with references to itself. There are cases where you'd intentionally want to visit the same node multiple times, such as traversing a directed acyclic graph (which would work just fine and eventually terminate, but would visit a node once for each parent it has connected to it) - but it's up to the user to be sure you don't create infinite loops.
+
+### isTraversable
+A default check if something can be traversed - currently it is arrays and plain objects.
+
+### traverse
+The default traversal function used in other tree methods if you don't supply one. It returns `_.values` if it has any or the passed in value if doesn't
```

Comment:
🤔 is `_.values` right here? the implementation doesn't have it `export let traverse = x => isTraversable(x) && !_.isEmpty(x) && x`